source/directives/uiTreeNode.js

```diff
@@ -572,6 +572,12 @@
                 }
               };
             })();
+            
+            keydownHandler = function (e) {
+              if (e.keyCode == 27) {
+                dragEndEvent(e);
```

Comment:
Perfect. So we shouldn't have called the `dragEnd` method directly, but instead call `dragEndEvent` first?